crates/project/src/lsp_command.rs

```diff
@@ -3178,3 +3183,477 @@ impl LspCommand for LinkedEditingRange {
         BufferId::new(message.buffer_id)
     }
 }
+
+impl GetDocumentDiagnostics {
+    pub fn deserialize_lsp_diagnostic(diagnostic: proto::LspDiagnostic) -> Result<lsp::Diagnostic> {
```

Comment:
```suggestion
    fn deserialize_lsp_diagnostic(diagnostic: proto::LspDiagnostic) -> Result<lsp::Diagnostic> {
```

Same for serialization.
Is there a point in these impls being separate, or could they be moved entirely into the `impl LspCommand for GetDocumentDiagnostics`?

Also, somewhat surprised that we could not reuse the old code for that, we sure did get diagnostics pushed from LSP before, what is the issue with that?

Reply:
>Same for serialization.

Done. Thanks!

>Is there a point in these impls being separate, or could they be moved entirely into the impl LspCommand for GetDocumentDiagnostics?

I'm not sure if I fully understand that. Don't we need to define them in the `LspCommand` trait first?